doc/debugging.md

```diff
@@ -99,3 +99,32 @@ Also to inspect logs in a more flexible manner, instead of the strict output fro
 adb logcat | grep 'ReactNativeJS\|StatusModule\|GoLog'
 ```
 
+### From @clauxx (visual debugging):
+
+When building/fixing/refactoring components, it can be annoying to set-up the component for instant visual feedback (i.e. finding it in the UI when fixing/refactoring; finding a place for it when building from scratch). Very often if the location of the screen containing your component nested deeply in the app, you might have to do a lot of repeated navigation, nevermind logging in every time the app reloads (even with fast refresh disabled this happens quite often).
+
+To aleviate this discomfort and make the development cycle more *pleasant*, you can use one of the following options (*_both of which involve the ... REPL ༼ つ ╹ ╹ ༽つ, so make sure you have that set-up with your IDE/editor_*)
+
+#### REPL-ing to a screen
+This one is straightforward, just evaluate a navigation dispatch form from anywhere:
+```cljs
+(comment (rf/dispatch [:navigate-to :your-favorite-buggy-screen]))
+```
+> TIP: you can run this from an untracked user namespace `(ns user)`, from which you can experiment or interact with the REPL. This is where I usually keep my development navigation forms (thanks to @ilmotta for both these tips).
+
+#### REPL-ing a component
+This was is also straighforward, but there are some small differences. Just like above, we only have to evaluate a dispatch form, as follows:
+```cljs
+;; example debugging a quo component, although it could be any other component e.g. from the `status-im2` ns.
+(comment (re-frame/dispatch [:debug-component
+                      [quo/slide-button
+                       {:track-icon          :face-id
+                        :track-text          ":debug-component example"
+                        :customization-color :blue
+                        :on-complete         identity}]]))
+```
+When evaluated, a full-screen bottom sheet will appear with your component inside. You can make changes to the props or the component internals and re-evaluate the changed forms, after which the component will be re-mounted without running a reload (unless you save the file). Other times it's quicker to just hit save and the component will be reloaded as usual (except for the props you evaluated, which have to be re-evaluated when changed).
+
+> TIP: you can have multiple versions of the component as separate comments to check for the component variations
+
+> UP-FOR-DISCUSSION: would be cool to leave comments with component usages at the bottom of the component file, especially where the props are more complex e.g. in most of `status-im` components. I saw/read this was somewhat common with clojure(script) and I feel like it would be useful in this case, but would rather have the more experienced clojure devs who maybe tried it in the past chip in on this.
```

Comment:
Interested in your opinions on this ☝️

Reply:
I agree, that's one of the reasons I suggested using StoryBook: It has this documentation feature. Not sure ibn our case how would be better to include that

Reply:
Leaving one or more `comment` macro blocks at the bottom of files is quite common in Clojure projects, but for the Quo library namespaces I think this is largely unnecessary because we have the component tests and the preview namespaces.

We will add Malli to self document components, so having yet another mechanism to see how to call components seems redundant to me.

All calls to `(h/render [component ...args])` in component tests can be eval'ed too and `h/query*` functions can be called to inspect results without the UI.

What would the bottom sections add in terms of value that can't be well explained in component tests?

Reply:
So for quo components I agree It to be mostly useless. I meant more in the context of status-im components, which have no previews. Tests do help, but the comments would allow to quickly see the component and interact with it outside the screen it resides in. Otherwise you'd have to navigate in the app and maybe put the screen in a specific state to see the component you need.

Reply:
Agreed @clauxx, that's why I made this comment https://github.com/status-im/status-mobile/pull/17729#discussion_r1370053205, which I think is a good usage of a `comment` macro. You may find this interesting (I'm not proposing it though) https://clojureverse.org/t/hyperfiddle-rcf-turn-your-rich-comment-forms-into-tests/7866